session.go

```diff
@@ -133,6 +138,20 @@ func WithMetadata(meta string) ConnectOption {
 	}
 }
 
+// WithUserAgent configures the user-agent string for this session.
+// This is a means for you to identify your application to ngrok and to
+// differentiate it from other applications using the agent library.
+//
+// We recommend to set this with your application name and version following
+// the format specified in RFC 7231. See [MDN's user-agent documentation] for examples.
```

Comment:
I'm not 100% sure on the recommendation here. Currently, this doesn't show up in the dashboard at all, right, so this is really just currently useful for usage metrics.

For usage metrics, we'd only want unique user agents for quite large buckets (so i.e. a library using this and expecting other users of it), but not for individual applications (like a once-off webhook test, if I set that to 'my-unpublished-testing-app', that actually gives less information than the default 'ngrok-go' one).

I actually kinda think the right recommendation might be "This value should typically be left unset. If you are creating a library or application you intend to publish broadly, setting a UserAgent may make sense. (etc etc RC 7231)"

.... Of course, the above is based on UserAgent being the unpublished internal-only thing.

I could also see us changing it so "UserAgent" is exposed on the session and we encourage changing it, and "Transport" gets updated to match what UserAgent is now (i.e. `ngrok/2` for ngrok agents, `ngrok-go/0` for ngrok-go, etc etc), and that would then give us information on library usage, while still letting people tweak user agents.

Reply:
i think eventually people will want to be able to see application names in the dashboard and API so they can quickly see which things are run by which app without having to divine this by correlating IPs or URLs or whatever. what that looks like is very TBD!